### PR TITLE
mkudffs workarounds for Windows chkdsk bugs

### PR DESCRIPTION
**Pull request by smagnani**
For upstream consideration. This is my first attempt at a GitHub pull request, let me know if I've botched something.

### CODE REVIEW

File: mkudffs/mkudffs.c [hunk 1]
```diff
@@ -403,6 +403,11 @@ void split_space(struct udf_disc *disc)
 				size = ext->blocks;
 			}
 
+			// Make sure partition doesn't have a space bitmap with a partial last byte.
+			// It's a corner case that some systems (Windows chkdsk) don't handle properly.
+			if ((disc->flags & FLAG_UNALLOC_BITMAP) && (size & 7))
+				size &= ~7;
```
Comment (pali) [inline mkudffs/mkudffs.c:409]: Is this problem only for unallocated bitmap? Or also with freed bitmap?

Comment (smagnani) [reply mkudffs/mkudffs.c:409]: Yes; when I try to run chkdsk on a partition formatted with --space=freedbitmap I get this:

`Only unallocated space bitmap is supported for free space
management in UDF. Chkdsk aborted.`

Comment (pali) [reply mkudffs/mkudffs.c:409]: Ok. I'm fine with this one commit. But github UI does not allow me to merge just one commit...

Comment (pali) [reply mkudffs/mkudffs.c:409]: So move other commit into separate pull request and so I could merge this one.

Comment (pali) [reply mkudffs/mkudffs.c:409]: I cherry-picked manually this one commit into master branch.

**Comment by pali** [comment]
Second commit was already merged into master. About first commit I do not think that it is a good idea to merge it. If chkdsk tool is not able to correctly detect fs then it really should not touch such fs. So adding hacks into mkfs just for broken chkdsk can lead to more issues. chkdsk/fsck should fix broken fs and not let it in broken state after successfull run. So I'm closing this pull request.

**Pull request closed without merging**